server/models/attendanceSettingsModel.js

```diff
@@ -0,0 +1,58 @@
+import mongoose from 'mongoose'
+import AppError from "../AppError";
+import {userHasRole} from "../utils";
+import * as SC from "../serverconstants";
+import * as EC from "../errorcodes";
+import {isAuthenticated, isAdmin, isSuperAdmin, hasRole} from "../utils"
+
+mongoose.Promise = global.Promise
+
+let attendanceSettingsSchema = mongoose.Schema({
+
+    minFullDayHours:{type:Number, default:7},
+    minHalfDayHours:{type:Number, default:4},
+    dayStartTime:{type:String, default:"10:00"},
+    dayEndTime:{type:String, default:"22:00"}
+
+})
+
+/*I am hoping that default add / update / delete of super will work here */
+
+attendanceSettingsSchema.statics.getAdminAttendanceSettings = async (loggedInUser) => {
+
+    if (isAdmin || isSuperAdmim) {
```

Comment:
isAdmin and isSuperAdmin are methods and needs koa context (isAdmin(ctx)), please us hasRole here

Reply:
Also I don't think super admin has any business here as super admin should not be involved in business. It is just there to manage things like roles/permissions/users etc

Reply:
But i have seen that isADmin and isSuperAdmin are used like that ? Anyway i will change.

I thought only SuperAdmin and Admin both will be able to see attendance settings. Anyway I think you are right superadmin is just above all and he don't have anything with it. Will update.